stage2/BuildBuild.scala

```diff
@@ -29,17 +29,26 @@ class BuildBuild(context: Context) extends BasicBuild(context){
                 None
               else Some{
                 val checkoutDirectory = new GitDependency(base, hash).checkout
-                val build = new BasicBuild( context.copy( projectDirectory = checkoutDirectory ++ "/nailgun_launcher" ) )
-                val cl = build
-                  .classLoader(classLoaderCache)
                 // Note: cbt can't use an old version of itself for building,
                 // otherwise we'd have to recursively build all versions since
                 // the beginning. Instead CBT always needs to build the pure Java
                 // Launcher in the checkout with itself and then run it via reflection.
-                cl
-                  .loadClass( "cbt.NailgunLauncher" )
-                  .getMethod( "getBuild", classOf[AnyRef] )
-                  .invoke( null, managedContext.copy(cbtHome=checkoutDirectory) )
+                val proj = new File(projectDirectory.toString.dropRight("/build".length))
+                val process = new ProcessBuilder(checkoutDirectory.toString + "/cbt", "direct", "serializeDependencies")
+                  .directory(proj)
+                  .start
+                process.waitFor
+                import java.io.{BufferedReader, InputStreamReader}
+                import collection.JavaConverters._
+                val buildContext = context.copy(projectDirectory = proj, cbtHome = checkoutDirectory)
+                val depString = new BufferedReader( new InputStreamReader( process.getInputStream )).readLine
+                val deps = depString.drop("Vector(".length).dropRight(1).split(",").map(dependency => dependency match {
+                  case d if d.contains(':') => MavenDependency deserialize d
+                  case d                    => new BuildDependency(buildContext.copy( projectDirectory = new File(d) ))
+                })
+                val build = new BasicBuild(buildContext) 
+                build.dependencies :+ deps
```

Comment:
Actually you should go a step further. The code that computes `deps` should be in the `def dependencies there`. And you should do `new BuildInterface{ ... }`, implementing all methods as forwarders to the other cbt process. Then the build truely runs on another jvm, not only the dependencies task. Depending on how you do it, every task would run in it's own JVM, which is probably super slow, but let's start with that. After that we can consider maybe spinning up a second nailgun server under a different port to make it practical. But we'll see